toolkit/docs/building/building.md

```diff
@@ -1,258 +1,358 @@
+Note that the image config file passed to the `CONFIG_FILE` option _only_ builds the packages included in the image config, plus all packages needed to build/run those packages. In other words, rather than building the entirety of the CBL-Mariner repo, only the packages needed by the image config and their dependencies will be built.
+
+#### Targeted Package Building
+
+Beginning with the `CBL-Mariner 2.0`'s 2022 October Release (`2.0.20221007`), it is possible to rapidly build one or more packages "in-tree". This technique can be helpful for modifying an existing SPEC file or adding a new one to CBL-Mariner.
 
-### **Targeted Package Building**
-Beginning with the CBL-Mariner 2.0's 2022 October Release (2.0.20221007) it is possible to rapidly build one or more packages "in-tree".  This technique can be helpful for modifying an existing SPEC file or adding a new one to CBL-Mariner.
+We do this by limiting the packages the toolkit knows about by only packaging the desired SRPMS via `SRPM_PACK_LIST`. Also, recall that individual packages can be developed outside the core repo; refer to the [CBL-MarinerTutorials](https://github.com/microsoft/CBL-MarinerTutorials) repo for details.
+
+|Argument|Use|
+|:-|-|
+| SRPM_PACK_LIST | Only package these `.spec` files for further processing |
+| PACKAGE_BUILD_LIST | Build this package, and any dependencies we know about **if they are not already built**|
+| PACKAGE_REBUILD_LIST | **ALWAYS** rebuild this package |
 
 ```bash
+# Clear any existing packages that the tools might know about
+sudo make clean-input-srpms clean-expand-specs
+
 # Build targeted packages
-sudo make build-packages -j$(nproc) REBUILD_TOOLS=y SRPM_PACK_LIST="openssh"
+packages="openssh at"
+sudo make build-packages -j$(nproc) QUICK_REBUILD=y SRPM_PACK_LIST="$packages"
 ```
-Note that this process will download dependencies from packages.microsoft.com and rebuild just the SPEC files indicated by the SRPM_PACK_LIST
 
-After building a package you may choose to rebuild it or build additional packages.  The optional `REFRESH_WORKER_CHROOT=n` option (default is `y`) will avoid rebuilding the worker chroot saving some additional build overhead
+Note that this process will download dependencies from [packages.microsoft.com](packages.microsoft.com) and rebuild just the `.spec` files indicated by the `SRPM_PACK_LIST`
+
+After building a package, you may choose to rebuild it or build additional packages.
 
 ```bash
-# Clean and rebuild targeted packages
+# Clean and rebuild targeted packages. Only rebuild the packages if they are not already built.
 sudo make clean-build-packages
-sudo make build-packages -j$(nproc) REBUILD_TOOLS=y SRPM_PACK_LIST="at openssh" REFRESH_WORKER_CHROOT=n
+sudo make build-packages -j$(nproc) QUICK_REBUILD=y CONFIG_FILE="" SRPM_PACK_LIST="at openssh"
```

Comment:
Edge case that I hope to make go away soon anyways so point will be moot. tldr: CONFIG_FILE="..." tells the tools to also build packages for that image definition. It defaults to core-efi.json, which means a bunch of extra stuff gets built. Because we used `SRPM_PACK_LIST` in the first one there is no way for the tools to build other packages to begin with, so it doesn't matter.

Soon I"m going to swap the default to be CONFIG_FILE="".